metadata-ingestion/src/datahub/ingestion/source/powerbi/powerbi.py

```diff
@@ -270,33 +283,37 @@ def transform_tags(tags: List[str]) -> GlobalTagsClass:
 
     def to_datahub_chart_mcp(
         self,
-        tile: PowerBiAPI.Tile,
+        tile: powerbi_data_classes.Tile,
         ds_mcps: List[MetadataChangeProposalWrapper],
-        workspace: PowerBiAPI.Workspace,
+        workspace: powerbi_data_classes.Workspace,
     ) -> List[MetadataChangeProposalWrapper]:
         """
         Map PowerBi tile to datahub chart
         """
-        logger.info("Converting tile {}(id={}) to chart".format(tile.title, tile.id))
+        logger.info(f"Converting tile {tile.title}(id={tile.id}) to chart")
         # Create a URN for chart
         chart_urn = builder.make_chart_urn(
             self.__config.platform_name, tile.get_urn_part()
         )
 
-        logger.info("{}={}".format(Constant.CHART_URN, chart_urn))
+        logger.info(f"{Constant.CHART_URN}={chart_urn}")
 
         ds_input: List[str] = self.to_urn_set(ds_mcps)
 
-        def tile_custom_properties(tile: PowerBiAPI.Tile) -> dict:
-            custom_properties = {
-                "datasetId": tile.dataset.id if tile.dataset else "",
-                "reportId": tile.report.id if tile.report else "",
-                "datasetWebUrl": tile.dataset.webUrl
-                if tile.dataset is not None
-                else "",
+        def tile_custom_properties(tile: powerbi_data_classes.Tile) -> dict:
+            custom_properties: dict = {
                 "createdFrom": tile.createdFrom.value,
             }
 
+            if tile.dataset_id is not None:
+                custom_properties["datasetId"] = tile.dataset_id
+
+            if tile.dataset is not None:
```

Comment:
should we also check tile.dataset.webUrl?

Reply:
done